File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int k = 1;
        int indicator = false;

        for( int i = 1; i < nums.size(); i++ ){
            if( nums.at(i) != nums.at(i-1) ){
                indicator = false;
                nums.at(k) = nums.at(i);
                k++;
            } else if( indicator == true ) {
                continue;
            } else if( indicator == false ){
                indicator = true;
                nums.at(k) = nums.at(i);
                k++;
            }
        }
        return k;
    }
};

File: 80-remove-duplicates-from-sorted-array-ii/remove-duplicates-from-sorted-array-ii.py
class Solution(object):
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        i, n = 0, len(nums)

        for j in range(n):
            if i < 2 or nums[j] != nums[i-2]:
                nums[i] = nums[j]
                i += 1
        return i
        
        